pkg/v1/webhooks/certificates.go

```diff
@@ -122,3 +135,30 @@ func ValidateTLSSecret(tlsSecret *corev1.Secret, certGracePeriod time.Duration)
 	}
 	return nil
 }
+
+// Write secret to persistent storage
+func updatePersistentTLSSecret(ctx context.Context, clusterClient client.Client, tlsSecret *corev1.Secret) error {
+	// if tlsSecret is nil nothign to do
+	if tlsSecret == nil {
+		return nil
+	}
+	currentSecret := &corev1.Secret{}
+	err := clusterClient.Get(ctx, client.ObjectKey{
+		Namespace: tlsSecret.Namespace,
+		Name:      tlsSecret.Name}, currentSecret)
+	if apierrors.IsNotFound(err) {
+		err = clusterClient.Create(ctx, tlsSecret)
+		if err != nil {
+			return err
+		}
+	}
+	if err != nil {
+		return err
+	}
+
+	err = clusterClient.Update(ctx, tlsSecret)
```

Comment:
question: do we want to do `retry.RetryOnConflict()` Or let caller to handle the errors such as "the object has been modified, please get the latest and try again"

Reply:
Done.